### PR TITLE
New lint: precedence, see issue #41

### PR DESCRIPTION
**Pull request by llogiq**

### CODE REVIEW

File: src/misc.rs [hunk 1]
```diff
@@ -169,3 +169,47 @@ impl LintPass for FloatCmp {
...
+	fn get_lints(&self) -> LintArray {
+        lint_array!(PRECEDENCE)
+	}
+	
+	fn check_expr(&mut self, cx: &Context, expr: &Expr) {
+		if let ExprBinary(Spanned { node: op, ..}, ref left, ref right) = expr.node {
+			if is_bit_op(op) {
+				if let ExprBinary(Spanned { node: lop, ..}, _, _) = left.node {
+					if is_arith_op(lop) {
+						cx.span_lint(PRECEDENCE, expr.span, "Operator precedence can trip the unwary. Please consider adding parenthesis to the subexpression to make the meaning more clear.");
```
Comment (Manishearth) [inline src/misc.rs]: Please try to keep line lengths less than 120, preferably less than 80 :)

File: src/misc.rs [hunk 2]
```diff
@@ -169,3 +169,47 @@ impl LintPass for FloatCmp {
...
+					}
+				} else {
+					if let ExprBinary(Spanned { node: rop, ..}, _, _) = right.node {
+						if is_arith_op(rop) {
+							cx.span_lint(PRECEDENCE, expr.span, "Operator precedence can trip the unwary. Please consider adding parenthesis to the subexpression to make the meaning more clear.");
```
Comment (Manishearth) [inline src/misc.rs]: ditto

**Comment by llogiq** [comment]
Sorry, will refactor with the next PR. Ok?

**Comment by Manishearth** [comment]
Yep.

-Manish Goregaokar

**Pull request merged**